packages/aws-rfdk/lib/deadline/lib/secrets-management.ts

```diff
@@ -0,0 +1,262 @@
+            //   <SUBNET_ID>,<ROLE>,<REGISTRATIONSTATUS>
+            .map(subnetRegistrationEntry => {
+              const [subnetID, registrationSettingEffect] = subnetRegistrationEntry;
+              return [
+                subnetID,
+                registrationSettingEffect.role.toString(),
+                (registrationSettingEffect.registrationStatus).toString(),
+              ].join(',');
+            })
+            // quote
+            .map(joinedSubnetArgValue => `--source-subnet "${joinedSubnetArgValue}"`),
+        );
+      },
+    });
+
+    // We can't use ScriptAsset.executeOn(...) because we need to run as "ec2-user".
+    // This is because Repository.configureClientInstance(...) used above will store the credentials
+    // in a per-user credential store that is only available to "ec2-user".
+    props.deploymentInstance.userData.addCommands(
+      `sudo --login -u ec2-user ${localScriptFile} ` + Fn.join(
+        ' ',
+        scriptArgs,
+      ),
+    );
+  }
+
+  private preparePythonScript(props: SecretsManagementIdentityRegistrationProps) {
+    const script = new ScriptAsset(this, 'ConfigureIdentityRegistrationSettingScript', {
+      path: path.join(
+        __dirname,
+        '..',
+        'scripts',
+        'python',
+        'configure_identity_registration_settings.py',
+      ),
+    });
+
+    // Grant access to ec2:DescribeSubnets. Subnet IPv4 CIDR ranges are not exposed through
+    // CloudFormation attributes. Instead, we must query them using the EC2 API on the deployment instance
+    props.deploymentInstance.grantPrincipal.addToPrincipalPolicy(new PolicyStatement({
+      actions: ['ec2:DescribeSubnets'],
+      // ec2:DescribeSubnets does not support resource level permissions. See
+      // https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html
+      resources: ['*'],
+    }));
+
+    this.adminCredentials.grantRead(props.deploymentInstance);
+
+    script.grantRead(props.deploymentInstance);
+    const localScriptFile = props.deploymentInstance.userData.addS3DownloadCommand({
+      bucket: script.bucket,
+      bucketKey: script.s3ObjectKey,
+      localFile: '/home/ec2-user/configure_identity_registration_settings.py',
+    });
+    props.deploymentInstance.userData.addCommands(
+      `chmod +x ${localScriptFile}`,
+      `chown ec2-user:ec2-user ${localScriptFile}`,
+    );
+    return localScriptFile;
+  }
+
+  private installDeadlineClient(props: SecretsManagementIdentityRegistrationProps) {
+    props.version.linuxInstallers.client.s3Bucket.grantRead(
+      props.deploymentInstance,
+      props.version.linuxInstallers.client.objectKey,
+    );
+    const clientInstallerPath = props.deploymentInstance.userData.addS3DownloadCommand({
+      bucket: props.version.linuxInstallers.client.s3Bucket,
+      bucketKey: props.version.linuxInstallers.client.objectKey,
+    });
+    props.deploymentInstance.userData.addCommands('set -x');
```

Comment:
I've confirmed this does not by reviewing CloudWatch logs. Our repository direct connection script runs in a bash sub-process and that securely handles the password.

Also, the SM admin password used by `packages/aws-rfdk/lib/deadline/scripts/python/configure_identity_registration_settings.py` is put in the `DL_SM_PW` environment variable and we use the `--password env:DL_SM_PW` method to avoid passing the password with command-line arguments.